packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -1425,6 +1429,40 @@ class OffsetLayer extends ContainerLayer {
       scene.dispose();
     }
   }
+
+  /// Capture an image of the current state of this layer and its children.
+  ///
+  /// The returned [ui.Image] will be offset by the top-left corner of [bounds],
+  /// and have dimensions equal to the size of [bounds] multiplied by [pixelRatio].
+  ///
+  /// Unlike [toImage], this creates a GPU-resident texture which has a backend
+  /// specific byte format. Rasterization will begin immediately on the engine
+  /// thread.
+  ///
+  /// The [pixelRatio] describes the scale between the logical pixels and the
+  /// size of the output image. It is independent of the
+  /// [dart:ui.FlutterView.devicePixelRatio] for the device, so specifying 1.0
+  /// (the default) will give you a 1:1 mapping between logical pixels and the
+  /// output pixels in the image.
+  ///
+  /// See also:
+  ///
+  ///  * [RenderRepaintBoundary.toImage] for a similar API at the render object level.
+  ///  * [dart:ui.Scene.toGpuImage] for more information about the image returned.
+  ui.Image toGpuImage(Rect bounds, { double pixelRatio = 1.0 }) {
```

Comment:
Why not provide a full transform so that this sub-scene can be rasterized with all inherited transforms. The existing scene should already contain the pixelRatio in a transform inserted early on in the scene building process.

Pixel ratio is the primary concern here for what scale to render it with, but if some sort of scene magnification is in effect you will lose that information by only adjusting for the pixel ratio.

Reply:
Depending on how this is constructed, you could always provide an arbitrary transform by embedding said transform in your scene. If A widget/RO that uses this API is not aware of a parent transform, or aware of changes in parent transform, then there may be pixelation issues, but that is the same caveat that the existing async API has